gui/wxpython/datacatalog/tree.py

```diff
@@ -954,19 +962,27 @@ def InsertLayer(self, name, mapset_node, element_name):
         self._model.SortChildren(mapset_node)
         self.RefreshNode(mapset_node, recursive=True)
 
-    def InsertMapset(self, name, location_node):
+    def InsertMapset(self, name, location_node, grassdb_node):
         """Insert new mapset into model and refresh tree.
         Assumes mapset is empty."""
+        mapset_path = os.path.join(grassdb_node.data['name'],
+                                   location_node.data['name'],
+                                   name)
         mapset_node = self._model.AppendNode(parent=location_node,
-                                             data=dict(type="mapset", name=name))
+                                             data=dict(type='mapset',
+                                                       name=name,
+                                                       current=False,
+                                                       lock=is_mapset_locked(mapset_path),
+                                                       is_different_owner=is_different_mapset_owner(mapset_path),
+                                                       owner=get_mapset_owner(mapset_path)))
         self._model.SortChildren(location_node)
         self.RefreshNode(location_node, recursive=True)
         return mapset_node
 
     def InsertLocation(self, name, grassdb_node):
         """Insert new location into model and refresh tree"""
         location_node = self._model.AppendNode(parent=grassdb_node,
-                                               data=dict(type="location", name=name))
+                                               data=dict(type='location', name=name))
```

Comment:
Don't bother changing quotes from `"` to `'`. Although most prevalent style in GRASS is now `'` for keys, the Black style, which we will use in the future, uses `"` for everything.

Reply:
For now, I would let it like it is in the last commit. At least it has a uniform form.